src/dsql/DdlNodes.epp

```diff
@@ -2941,7 +2941,7 @@ bool CreateAlterProcedureNode::executeAlter(thread_db* tdbb, DsqlCompilerScratch
 				if (secondPass)
 				{
 					P.RDB$PROCEDURE_TYPE.NULL = FALSE;
-					P.RDB$PROCEDURE_TYPE = (USHORT) prc_selectable;
+					P.RDB$PROCEDURE_TYPE = returns.getCount() != 0 ? (USHORT) prc_selectable : (USHORT) prc_executable;
```

Comment:
Are you sure that procedure without output parameters cannot be selectable?
`select count(*) from anyUDR()`

Reply:
>  `select count(*) from anyUDR()`

This causes ` SQL error code = -84; procedure TEST_UDR does not return any values`. Even if the procedure has `prc_selectable` type.

Reply:
This solves a completely different problem. https://github.com/FirebirdSQL/firebird/issues/7699
As for creating external procedures and functions in one-pass, I agree with @asfernandes. This is not necessary and can even be harmful in some cases.

Reply:
@sim1984 yes, this is exactly the bug we were fixing. I suggested a single pass extraction as an extension to the original patch.